src/util/String.cc

```diff
@@ -171,4 +173,20 @@ string String::fromHex(string const& hex, string const& prefix) {
     return out;
 }
 
+string String::toLower(string const& str) {
+    string result = str;
+    transform(result.begin(), result.end(), result.begin(), ::tolower);
+    return result;
+}
+
+string String::toUpper(string const& str) {
+    string result = str;
+    transform(result.begin(), result.end(), result.begin(), ::toupper);
+    return result;
+}
+
+bool String::equalIgnoreCase(std::string const& str1, std::string const& str2) {
+    return toLower(str1) == toLower(str2);
```

Comment:
How about `std::equal(str1.begin(), str1.end(), str2.begin(), str2.end(), ichar_equals);`?

Reply:
That implies adding the comparator function, something along these lines:
```
bool ichar_equals(char a, char b)
{
    return std::tolower(static_cast<unsigned char>(a)) ==
           std::tolower(static_cast<unsigned char>(b));
}
```
Which is a reasonable option too.